lib/Core/Server/LanguageServer.php

```diff
@@ -223,13 +221,7 @@ private function dispatchRequest(MessageTransmitter $transmitter, Dispatcher $di
                 $response = yield $dispatcher->dispatch($request);
             } catch (ExitSession $e) {
                 $connection->stream()->end();
-
-                if ($this->streamProvider instanceof ResourceStreamProvider) {
-                    throw new ShutdownServer(
-                        'Exit called on STDIO connection, exiting the server'
-                    );
-                }
-                return;
+                exit;
```

Comment:
Sorry, I meant to exit the _loop_ and pass back to control to the thing which started the server (e.g. the command, which would then exit with 0 for example).

At least this could be done in `StartCommand` or it could be handled here (i think): 
https://github.com/phpactor/language-server/blob/3475c59d76d8f5a7eb3d8745387f734dc057c8e4/lib/Core/Server/LanguageServer.php#L106

Perhaps calling `Loop::stop()` on a ShutdownException would be enough (needs to be tested though)

Reply:
✓ updated

> Perhaps calling Loop::stop() on a ShutdownException would be enough (needs to be tested though),

Seems to be working.